crates/core/tedge/src/bridge/config.rs

```diff
@@ -91,17 +101,19 @@ impl BridgeConfig {
         Ok(())
     }
 
-    pub fn validate(&self) -> Result<(), ConnectError> {
+    pub fn validate(&self, use_basic_auth: bool) -> Result<(), ConnectError> {
         if !self.bridge_root_cert_path.exists() {
             return Err(ConnectError::Certificate);
         }
 
-        if !self.bridge_certfile.exists() {
-            return Err(ConnectError::Certificate);
-        }
+        if !use_basic_auth {
+            if !self.bridge_certfile.exists() {
+                return Err(ConnectError::Certificate);
+            }
 
-        if !self.bridge_keyfile.exists() {
-            return Err(ConnectError::Certificate);
+            if !self.bridge_keyfile.exists() {
+                return Err(ConnectError::Certificate);
+            }
         }
```

Comment:
No validation required when `use_basic_auth == true`? Like whether the `credentials` file exists?

This validation is done during the `Validating the bridge certificates.` phase of `tedge connect`. We may wanna change that message as well, to better align with the new option.